src/aggregator/aggregator/generic_elem.go

```diff
@@ -505,31 +494,49 @@ func (e *GenericElem) Consume(
 	return canCollect
 }
 
-func (e *GenericElem) newFlushStateWithLock(agg timedAggregation) aggFlushState {
-	// note: flushState might be empty for the first flush
-	flushState := e.flushState[agg.startAtNanos]
-	// copy the lockedAgg data to the flushState while holding the lock.
+func (e *GenericElem) isFlushed(c consumeState) bool {
+	return e.flushState[c.startAt].flushed
+}
+
+// append the consumeState for the timedAggregation to the provided slice if it matches the provided filter.
+// returns the updated slice and true if added.
+func (e *GenericElem) appendConsumeStateWithLock(
+	agg timedAggregation,
+	toConsume []consumeState,
+	filter func(consumeState) bool) ([]consumeState, bool) {
```

Comment:
Maybe name this `exclude` or something? Can be a bit confusing understanding if this filters in or out

Reply:
well it actually would be `include`, which is what a filter typically does. but point taken, will rename.

Reply:
Ah yeah you truncate the slice in the `!filter`

Reply:
yea this awkward `filter` function is just to support memory reuse.